File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        
        int dirX, dirY;
        dirX = 0;
        dirY = 1;
        
        int x, y;
        x = y = 0;
        
        for(int i = 0; i < instructions.size(); i++) {
            char curChar = instructions[i];
            int temp;
            
            if(curChar == 'L') {
                temp = dirY;
                dirY = dirX;
                dirX = -1*temp;
            } else if (curChar == 'R') {
                temp = dirX;
                dirX = dirY;
                dirY = -1*temp;
            } else {
                x+= dirX;
                y+= dirY;
            }
        }
        
        if (x == 0 && y == 0 || dirX != 0 || dirY != 1) {
            return true;
        }
        
        return false;
    }
};

File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        dirX, dirY = 0,1
        
        x,y = 0,0
        
        for step in instructions:
            if step == 'L':
                dirX,dirY = -1*dirY,dirX
            elif step == 'R':
                dirX,dirY = dirY,-1*dirX
            else:
                x+=dirX
                y+=dirY
        
        if (x,y) == (0,0) or dirX != 0 or dirY != 1:
            return True
        
        return False